ms/MSSel/MSFieldIndex.cc

```diff
@@ -132,14 +131,15 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
     else       reg=reg.fromPattern(pattern);
     
     //    cerr << "Pattern = " << pattern << "  Regex = " << reg.regexp() << endl;
-    IPosition sh(msFieldCols_p.name().getColumn().shape());
+    Vector<Bool> flagRow=msFieldCols_p.flagRow().getColumn();
+    Vector<String> codes=msFieldCols_p.code().getColumn();
+    IPosition sh(codes.shape());
```

Comment:
CODE and NAME are two columns of the FIELD sub-table and each NAME will have CODE cell associated with it.  So codes.shape() will be same as msFieldCols_p.name().getColumn().shape().

I made this change since, as you also note, it's the shape of the CODE column that really matters in the code later.  Also, in the earlier code the NAME column would be read unnecessarily to just get the shape.